server.go

```diff
@@ -336,3 +342,17 @@ func (sb *SplitBytes) Next() bool {
 func (sb *SplitBytes) Chunk() []byte {
 	return sb.currentChunk
 }
+
+func (s *Server) registerPlugin(p plugin) {
+	s.pluginMtx.Lock()
+	s.plugins = append(s.plugins, p)
+	s.pluginMtx.Unlock()
```

Comment:
I don't have a strong opinion since this isn't a performance sensitive place. We only flush every 10s so nbd…